src/components/LmsConfigurations/CornerstoneIntegrationConfigForm.jsx

```diff
@@ -0,0 +1,256 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import isEmpty from 'lodash/isEmpty';
+import {
+  ValidationFormGroup, Input, StatefulButton, Icon,
+} from '@edx/paragon';
+import { snakeCaseFormData } from '../../utils';
+import LmsApiService from '../../data/services/LmsApiService';
+import StatusAlert from '../StatusAlert';
+import SUBMIT_STATES from '../../data/constants/formSubmissions';
+import { handleErrors, validateLmsConfigForm } from './common';
+
+export const REQUIRED_CORNERSTONE_CONFIG_FIELDS = [
+  'cornerstoneBaseUrl',
+];
+
+function configFormReducer(state, action) {
+  switch (action.type) {
+    case 'PENDING': {
+      return { ...state, error: null, submitState: SUBMIT_STATES.PENDING };
+    }
+    case 'ERROR': {
+      return { ...state, error: action.value, submitState: SUBMIT_STATES.ERROR };
+    }
+    case 'INVALID': {
+      return {
+        ...state,
+        submitState: SUBMIT_STATES.ERROR,
+        invalidFields: action.value,
+        error: 'Form could not be submitted as there are fields with invalid values. Please correct them below.',
+      };
+    }
+    case 'COMPLETE': {
+      return {
+        ...state,
+        invalidFields: {},
+        error: null,
+        submitState: SUBMIT_STATES.COMPLETE,
+      };
+    }
+    case 'SET_CONFIG': {
+      return {
+        ...state,
+        config: action.value,
+      };
+    }
+    case 'BASE_SUBMIT': {
+      return {
+        ...state,
+        submitState: SUBMIT_STATES.DEFAULT,
+      };
+    }
+    default: {
+      return { state };
```

Comment:
Does this one not need a spread operator for some reason? (I could be wrong, I'm not amazing at React. yet.)

Reply:
That's for modifying an iterable isn't it? We're not modifying anything here.